tests/bluetooth/controller/ctrl_feature_exchange/src/main.c

```diff
@@ -0,0 +1,293 @@
+	zassert_equal(conn.lll.event_counter, 0,
+		      "Wrong event-count %d\n", conn.lll.event_counter);
+}
+
+/*
+ * +-----+ +-------+            +-----+
+ * | UT  | | LL_A  |            | LT  |
+ * +-----+ +-------+            +-----+
+ *   |        |                   |
+ *   |        |    LL_FEATURE_REQ |
+ *   |        |<------------------|
+ *   |        |                   |
+ *   |        | LL_FEATURE_RSP    |
+ *   |        |------------------>|
+ *   |        |                   |
+ */
+void test_feature_exchange_mas_rem(void)
+{
+	struct node_tx *tx;
+
+	struct pdu_data_llctrl_feature_req remote_feature_req;
+	struct pdu_data_llctrl_feature_rsp local_feature_rsp;
+
+	test_set_role(&conn, BT_HCI_ROLE_MASTER);
+	ull_cp_state_set(&conn, ULL_CP_CONNECTED);
+
+	event_prepare(&conn);
+
+	lt_tx(LL_SLAVE_FEATURE_REQ, &conn, &remote_feature_req);
+
+	event_done(&conn);
+
+	event_prepare(&conn);
+
+	lt_rx(LL_FEATURE_RSP, &conn, &tx, &local_feature_rsp);
+	lt_rx_q_is_empty(&conn);
+
+	event_done(&conn);
+
+	ut_rx_q_is_empty();
+	zassert_equal(conn.lll.event_counter, 2,
+		      "Wrong event-count %d\n", conn.lll.event_counter);
+}
+
+void test_feature_exchange_mas_rem_2(void)
+{
+	u64_t err;
+	struct node_tx *tx;
+	struct node_rx_pdu *ntf;
+
+	struct pdu_data_llctrl_feature_req remote_feature_req;
+	struct pdu_data_llctrl_feature_rsp local_feature_rsp;
+
+	test_set_role(&conn, BT_HCI_ROLE_MASTER);
+	ull_cp_state_set(&conn, ULL_CP_CONNECTED);
+
+	event_prepare(&conn);
+	lt_tx(LL_SLAVE_FEATURE_REQ, &conn, &remote_feature_req);
+	event_done(&conn);
+	err = ull_cp_feature_exchange(&conn);
+	zassert_equal(err, BT_HCI_ERR_SUCCESS, NULL);
+
+	event_prepare(&conn);
+	lt_rx(LL_FEATURE_RSP, &conn, &tx, &local_feature_rsp);
+	lt_rx_q_is_empty(&conn);
+	event_done(&conn);
+
+	ut_rx_pdu(LL_SLAVE_FEATURE_REQ, &ntf, &remote_feature_req);
+	ut_rx_q_is_empty();
+	zassert_equal(conn.lll.event_counter, 2,
+		      "Wrong event-count %d\n", conn.lll.event_counter);
+}
+
+
+void test_slave_feature_exchange_sla_loc(void)
+{
+	u64_t err;
+	struct node_tx *tx;
+	struct node_rx_pdu *ntf;
+
+	struct pdu_data_llctrl_feature_req local_feature_req;
+	struct pdu_data_llctrl_feature_rsp remote_feature_rsp;
+
+	test_set_role(&conn, BT_HCI_ROLE_SLAVE);
+	/* Connect */
+	ull_cp_state_set(&conn, ULL_CP_CONNECTED);
+
+	/* Initiate a Feature Exchange Procedure */
+	err = ull_cp_feature_exchange(&conn);
+	zassert_equal(err, BT_HCI_ERR_SUCCESS, NULL);
+
+	event_prepare(&conn);
+	/* Tx Queue should have one LL Control PDU */
+	lt_rx(LL_SLAVE_FEATURE_REQ, &conn, &tx, &local_feature_req);
+	lt_rx_q_is_empty(&conn);
+
+	/* Rx */
+	lt_tx(LL_FEATURE_RSP, &conn, &remote_feature_rsp);
+
+	event_done(&conn);
+	/* There should be one host notification */
+
+	ut_rx_pdu(LL_FEATURE_RSP, &ntf,  &remote_feature_rsp);
+	ut_rx_q_is_empty();
+	zassert_equal(conn.lll.event_counter, 1,
+		      "Wrong event-count %d\n", conn.lll.event_counter);
+}
+
+
+void test_feature_exchange_mas_loc_unknown_rsp(void)
```

Comment:
Not making sense; the LL_UNKNOWN_RSP is not valid for LL_FEATURE_REQ only for LL_SLAVE_FEATURE_REQ.

Reply:
changed accordingly